### PR TITLE
LES-41-gender

### PR DESCRIPTION
**Pull request by OxygenO22**

### CODE REVIEW

File: src/compopnents/radio/RadioGender.jsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                            Other
+                        </p>
+                    </div>
+                </div>
+            </label>
+        </div>
+    );
+}
+
+export default RadioGender;
```
Comment (Evgeniy121212) [inline src/compopnents/radio/RadioGender.jsx:58]: change to `selector` or `dropdown` component

Comment (Evgeniy121212) [inline src/compopnents/radio/RadioGender.jsx:58]: upgrade to .tsx

File: src/compopnents/radio/RadioGender.module.css [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+.radio__inner {
+    margin-right: 10px;
+}
+
+.radio__text {
+    font-size: 1.2em;
+    font-weight: 700;
+}
```
Comment (Evgeniy121212) [inline src/compopnents/radio/RadioGender.module.css:35]: good, just upgrade to .scss

**Review by Evgeniy121212 — COMMENTED** [review]
please, remove unnecessary files from that PR - it can affects other peoples

leave only `gender`-files (currently named RedioGender):
`src/compopnents/gender/Gender.tsx` (! implement .tsx)
`src/compopnents/gender/Gender.module.scss` (! change to .scss)

(and rename folder `src/compopnents` -> `src/components`)

**Comment by NatashaMikhaltsova** [comment]
please, remove unnecessary files (src/App.js, src/components/selector/Selector.module.css), classes should be used from Selector.module.scss

File: src/compopnents/radio/RadioGender.jsx (hunk 1 above)
Comment (NatashaMikhaltsova) [reply src/compopnents/radio/RadioGender.jsx]: @Evgeniy121212 Is it correct to rename component to Selector? May be GenderSelector will be better?

Comment (Evgeniy121212) [reply src/compopnents/radio/RadioGender.jsx]: yes, it better. agreed for `GenderSelector`

**Comment by OxygenO22** [comment]
Done, I deleted the unnecessary file;
Validation has not been done yet;

File: src/components/selector/Selector.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
+import React, {useState} from 'react';
+import classes from './Selector.module.scss'
+
+const Selector = (props) => {
+    const [radio, setRadio] = useState('');
+    /* const [radioDirty, setRadioDirty] = useState(false);
```
Comment (Terapool) [inline src/components/selector/Selector.tsx:6]: Please, remove commented code

File: src/components/selector/Selector.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+                            //onBlur={e => blurHandler(e)}
+                            {...props} 
+                        />
+                        <p className={classes.radio__text}>
+                            Other
+                        </p>
+                    </div>
+                </div>
+            </label>
+            {/* {(radioDirty && radioError) && <div style={{color: 'red'}}>{radioError}</div>} */}
```
Comment (Terapool) [inline src/components/selector/Selector.tsx:68]: Please, remove commented code

**Review by Terapool — CHANGES_REQUESTED** [review]

File: src/components/selector/Selector.tsx (hunk 4 above)
Comment (OxygenO22) [reply src/components/selector/Selector.tsx:68]: done

**Review by Terapool — APPROVED** [review]

**Review by NatashaMikhaltsova — CHANGES_REQUESTED** [review]

File: src/components/Selector.tsx [hunk 5]
```diff
@@ -0,0 +1,60 @@
+import React, {useState} from 'react';
+import classes from './Selector.module.scss';
+import '../index.scss';
+
+const Selector = (props) => {
```
[the remark this answers is not in the document]
Comment (OxygenO22) [reply src/components/Selector.tsx:5]: done

File: src/App.tsx [hunk 6]
```diff
@@ -1,24 +1,13 @@
...
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+      <form>
+            <Selector/>
+      </form>
+
```
[the remark this answers is not in the document]
Comment (OxygenO22) [reply src/App.tsx:10]: done

File: src/components/GenderSelector.tsx [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+            </label>
+            {/* <input className={classes.button}
+                    type="submit"
+                    value="Finish"
+                    onClick={e => {
+                        e.preventDefault();
+                        blurHandler();
+                    }} 
+                />  */} // for tests
+        </div>
```
Comment (NatashaMikhaltsova) [inline src/components/GenderSelector.tsx:86]: remove commented code

**Review by NatashaMikhaltsova — CHANGES_REQUESTED** [review]

File: src/components/GenderSelector.tsx (hunk 7 above)
Comment (OxygenO22) [reply src/components/GenderSelector.tsx:86]: done

**Review by DimitarShemet — APPROVED** [review]

File: src/components/GenderSelector.tsx (hunk 7 above)
Comment (NatashaMikhaltsova) [reply src/components/GenderSelector.tsx:86]: commented code is still present in src/components/GenderSelector.tsx

**Review by NatashaMikhaltsova — CHANGES_REQUESTED** [review]

File: src/components/GenderSelector.tsx [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+                        <div className={classes.radio__item}> 
+                            <input 
+                                className={classes.radio__inner}
+                                type='radio'
+                                name='Male'
+                                checked={radio === 'Male'}
+                                onChange={e => {
+                                    setRadio(e.target.name);
+                                    setRadioError('');
+                                }}
```
Comment (Kurilo-Pavel) [inline src/components/GenderSelector.tsx:34]: I think it's worth taking out the function and then passing it

**Review by Kurilo-Pavel — CHANGES_REQUESTED** [review]

**Review by NatashaMikhaltsova — APPROVED** [review]

**Review by valer-sky — APPROVED** [review]

**Pull request merged**